File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    dict = {}
    for n, y in zip(name, yearning):
        dict[n] = y
     
    for tc in photo:
        cnt = 0
        for p in tc:
            if p in name:
                cnt += dict[p]
        answer.append(cnt)
    
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    
    const scoreMap = new Map();
    
    for (let i = 0; i < name.length; i++) {
        scoreMap.set(name[i], yearning[i])
    }
    
    for (photo_name of photo) {
        var sum_score = 0
        for (n of photo_name) {
            if (scoreMap.get(n)) {
                sum_score += scoreMap.get(n)
            }
        }
        answer.push(sum_score)
    }    
    
    
    return answer;
}